ansible/all-in-one/configs/xsnippet-webserver-api.conf.j2

```diff
@@ -1,3 +1,24 @@
+{% if xsnippet_api_rate_limits %}
+# default limit for any request (per IP)
+limit_req_zone $binary_remote_addr zone=default:10m  rate=10r/s;
+
+# limit on creation of new snippets by unauthorized users (per IP).
+map "$http_authorization:$request_method" $unauthenticated_post_limit {
+    default     "";
+    ":POST"     $binary_remote_addr;
+}
+limit_req_zone $unauthenticated_post_limit zone=unauthenticated:10m rate=20r/m;
+
+# limit on creation of new snippets by authorized users (per IP).
```

Comment:
I think the limit by authorized users should be calculated based on JWT tokens, not on IP. Otherwise, the limit may apply to different users who are behind the same IP address. What do you think?